components/Article.tsx

```diff
@@ -67,6 +67,19 @@ const Article: React.FunctionComponent<Props> = props => {
             metaTags={metaTags}
             leftColumn={tocFragment}
         >
+            {/* Social image */}
+            {props.socialImage ? <meta name="twitter:card" content="summary_large_image" /> : ''}
+            {props.socialImage ? (
+                <meta name="twitter:image" content={props.socialImage} />
+                ) : (
+                    <meta name="twitter:image" content="https://learn.sourcegraph.com/headers/sourcegraph-learn-header.png" />
+                    )}
+            {props.socialImage ? (
+                <meta property="og:image" content={props.socialImage} />
+                ) : (
+                    <meta property="og:image" content="https://learn.sourcegraph.com/headers/sourcegraph-learn-header.png" />
+                    )}
+
```

Comment:
The meta tags should go into the `<head>` element along with the existing `og:image` and other meta tags in `Layout`

https://github.com/sourcegraph/learn/blob/1a22942c248e089c62c24544167fd3d9db966031/components/Layout.tsx#L67-L70

In that file we fall back to the default learn header if the page doesn't have its own social image so we can reuse `metaImage` for the `twitter:image`.

https://github.com/sourcegraph/learn/blob/1a22942c248e089c62c24544167fd3d9db966031/components/Layout.tsx#L46

Reply:
My concern with the current implementation is that it is not behaving as expected. The Twitter use case is the most important one.